File: solutions/sandbox/hypercube/hypercube.py
def pos(a):
	ans = 0
	n = length(ans)
	for i in range(length(a)):
		if a[n-i-1] == '1':
			ans = 2**(i+1)-1-ans

	return ans

n = int(input())
a, b = tuple(map(int, input().split()))

File: solutions/sandbox/hypercube/hypercube.cc
#include <bits/stdc++.h>

using namespace std;

int n;
string a, b;

int pos(string a) {

	long long ans = 0;
	for (int i=0; i<n; i++) {
		if (a[n-i-1] == '1') ans = (1 << i+1) -1 - ans;
	}

	return ans;
}

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);

	cin >> n >> a >> b;

	cout << pos(b)-pos(a)-1 << endl;

	return 0;
}
